features/frameworks/utest/source/utest_shim.cpp

```diff
@@ -123,7 +123,6 @@ static int32_t utest_us_ticker_run()
             callback();
         }
     }
-    return 0;
```

Comment:
No, the compilers (ARMCC, GCC and IAR) doesn't complain because this functions never returns.
Actually the compiler was complaining of the return statement because it can't be reached.